runtime/minor_gc.c

```diff
@@ -546,7 +545,8 @@ void caml_empty_minor_heap_promote(caml_domain_state* domain,
 
       for( r = ref_start ; r < foreign_major_ref->ptr && r < ref_end ; r++ )
       {
-        oldify_one (&st, **r, *r);
+        volatile value* pr = *r;
+        oldify_one (&st, *pr, pr);
```

Comment:
There is `value_ptr` which could be used here to make the replacement more obvious.

In fact, I wonder if it is not a case where `value_ptr` should replace `value *` further in the code base.

e.g. could it be that in `minor_gc.h`, the following:
```c
struct caml_ref_table CAML_TABLE_STRUCT(value *);
```
should be:
```c
struct caml_ref_table CAML_TABLE_STRUCT(value_ptr);
```
or something like that (with all the changes throughout the codebase that this would require)?

Reply:
(And, in this case, would it be a sign that we should inspect all the uses of `value*` in the runtime to see if they should be replaced with `value_ptr`? :)

Reply:
Often the elements of a `caml_table_ref` are manipulated without any risk of conflicting access, e.g., outside of minor collection remembered sets are domain-local. Making all these elements `volatile` would remove opportunities for compiler optimization for no good reason.

> (And, in this case, would it be a sign that we should inspect all the uses of `value*` in the runtime to see if they should be replaced with `value_ptr`? :)

I haven’t had a response on https://github.com/ocaml/ocaml/pull/12512, except from @gasche kindly trying to unstall the PR, so this auditing process is paused as far as I’m concerned.